src/check.rs

```diff
@@ -124,8 +125,15 @@ pub fn check(args: CheckArgs) -> i32 {
             let text_rules = text_rule_map(&rules, &settings);
             let ast_entrypoints = ast_entrypoint_map(&rules, &settings);
             let mut total_errors = 0;
-            for file in get_files(&args.files) {
-                match check_file(&path_rules, &text_rules, &ast_entrypoints, &file) {
+            for path in get_files(&args.files) {
+                // FIXME!!
+                let source = read_to_string(&path).unwrap();
```

Comment:
I'd prefer to fix this now rather than later. Maybe something like this:

```suggestion
            let err_source = "kj23brdf9kshb23owihqb";
            for path in get_files(&args.files) {
                let source = read_to_string(&path).unwrap_or(err_source.to_string());
                if source == err_source {
                    let file = SourceFileBuilder::new(path.to_string_lossy().as_ref(), "")
                        .finish();
                    let err_msg = format!("Failed to process: {}", path.display());
                    let violation = violation!(&err_msg);
                    let diagnostic = Diagnostic::new(&file, "E000", &violation);
                    println!("{}", diagnostic);
                    total_errors += 1;
                    continue;
                }
```
The diagnostic handling bit is very similar to a section lower down in this function, so should probably be pulled out into a new function.

Reply:
Yep, or pull the loop body out into a function that returns `Result`, and then can propagate the error up like that?

Reply:
Yes, I think that would be a better approach